app/models/short_url_request.rb

```diff
@@ -62,4 +69,8 @@ def strip_whitespace
     self.from_path = self.from_path.strip
     self.to_path = self.to_path.strip
   end
+
+  def should_republish_redirect?
+    accepted? && (from_path_changed? || to_path_changed?)
```

Comment:
I've rebased with some unit and acceptance tests and it does seem to work as intended.